scipy/optimize/__lbfgsb.c

```diff
@@ -2062,7 +2035,7 @@ formk(int n, int nsub, int* ind, int nenter, int ileave,
             for (jy = 0; jy < m - 1; jy++)
             {
                 js = m + jy;
-                temp_int = m - jy + 1;
+                temp_int = m - (jy + 1);
```

Comment:
Is there a reason for this line? It is an integer expression hence not sure what grouping does here.

Reply:
We'll shift the lower triangle part of the `m * m` top-left (line 2039) and bottom-right (line 2049) corner of `wn1` matrix of shape `2m * 2m`. Since the triangular part is shifted 1 row up and 1 column left, we'll be copying `m - (jy + 1)` elements where `jy`, the column index, starts from 0.

The reference implementation used expression `m - jy` because fortran counts from 1. (https://github.com/scipy/scipy/blob/adc4f4f7bab120ccfab9383aba272954a0a12fb0/scipy%2Foptimize%2Flbfgsb_src%2Flbfgsb.f#L1984-L1985)

Reply:
I am just too blind to see the obvious minus sign. Apologies for the noise.